server/src/main/java/io/javaclasses/filehub/user/CreateFolder.java

```
@@ -0,0 +1,58 @@
+package io.javaclasses.filehub.user;
+
+import com.google.errorprone.annotations.Immutable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A command to create a new folder.
+ *
+ * <p>This command provides information related to the folder being created. The information
+ * includes the folder name and the parent folder ID. The folder name must not be empty.
+ */
+@Immutable
+public final class CreateFolder extends AuthenticatedUserCommand {
+
+    private final String folderName;
+    private final FolderId parentFolder;
+
+    /**
+     * Initializes an instance of a command.
+     *
+     * @param folderName
+     *         the name of a new folder
+     * @param parentFolder
+     *         the identifier of the destination folder where the new directory will be created
+     * @throws InvalidFilenameException
+     *         if passed name for the folder is empty and cannot be used in the FileHub application
+     */
+    public CreateFolder(Token token, String folderName, FolderId parentFolder)
+            throws InvalidFilenameException {
+
+        super(token);
+        checkNotNull(folderName);
+        checkNotNull(parentFolder);
+
+        String trimmedFolderName = folderName.trim();
+        if (trimmedFolderName.isEmpty()) {
+            throw new InvalidFilenameException("Folder filename cannot be empty.");
```

`folder filename`? Maybe `folder name`? And if you throw an exception, it would be better to throw `InvalidFolderNameException`

---

Replaced with `folder name`.

By definition, a folder is a file. Although in our implementation these are different concepts, it makes no sense to separate the exceptions.